backoff/backoff.go

```diff
@@ -8,31 +8,31 @@ import (
 	"time"
 )
 
-// BackoffConfig configures a Backoff
-type BackoffConfig struct {
+// Config configures a Backoff
+type Config struct {
 	MinBackoff time.Duration `yaml:"min_period"`  // start backoff at this level
 	MaxBackoff time.Duration `yaml:"max_period"`  // increase exponentially to this level
 	MaxRetries int           `yaml:"max_retries"` // give up after this many; zero means infinite retries
 }
 
-// RegisterFlags for BackoffConfig.
-func (cfg *BackoffConfig) RegisterFlags(prefix string, f *flag.FlagSet) {
+// RegisterFlags for Config.
+func (cfg *Config) RegisterFlags(prefix string, f *flag.FlagSet) {
```

Comment:
[nit] By convention, we're used to call `RegisterFlagsWithPrefix()` the function to register flags with prefix. I would adhere here too.